### PR TITLE
Update plugin by removing `analytic` argument for device creation

### PR DESCRIPTION
**Pull request by antalszava**
Updates the plugin to cope with removing the `analytic` keyword argument as per https://github.com/PennyLaneAI/pennylane/pull/1079.

### CODE REVIEW

**Comment by josh146** [comment]
@thisac @trbromley just a reminder to review this one!

File: requirements.txt
Comment (thisac) [inline requirements.txt:3]: Do we want to wait until after release before merging this (and then update the requirements to the latest release instead)?

**Review by thisac — APPROVED** [review]
Looks good! Just one question: are the PL device tests updated as well, and should then the line `pl-device-test --device lightning.qubit --analytic False --skip-ops --shots=20000` in `build.yml` be changed here?

**Comment by josh146** [comment]
> Just one question: are the PL device tests updated as well, and should then the line pl-device-test --device lightning.qubit --analytic False --skip-ops --shots=20000 in build.yml be changed here?

Good catch! I'll change it and see

File: requirements.txt
Comment (josh146) [reply requirements.txt:3]: In the past we have, but we have so many plugins now, I think it is better to merge it as-is (and simply replace it with `pennylane>=0.15` once it is on PyPI).

By merging it in now, we can:

- Have its entry in the test matrix turn green,
- Keep track of it breaking (again!) between now and release :)

File: pennylane_lightning/lightning_qubit.py
Comment (trbromley) [inline pennylane_lightning/lightning_qubit.py:38]:
```suggestion
            the expectation values. Defaults to ``None`` if not specified. Setting
```

File: tests/test_apply.py
Comment (trbromley) [inline tests/test_apply.py:1137]: Why does this one need to change?

File: tests/test_expval.py
Comment (trbromley) [inline tests/test_expval.py:33]: Why do these ones need to change?

File: tests/test_var.py
Comment (trbromley) [inline tests/test_var.py:33]: Do we need to parametrize over `None` (which is the default)?

**Review by trbromley — CHANGES_REQUESTED** [review]
Thanks @antalszava! I had a few questions/comments, but otherwise looking good!

File: tests/test_apply.py
Comment (antalszava) [reply tests/test_apply.py:1137]: Because we implicitly had `analytic=True` before

File: tests/test_expval.py
Comment (antalszava) [reply tests/test_expval.py:33]: Same as above

File: tests/test_var.py
Comment (antalszava) [reply tests/test_var.py:33]: No, but in this PR rather changed the analytic argument rather than adjust how the testing is done

**Comment by antalszava** [comment]
Thanks for the comments and the catches! Updated the PR

File: tests/test_apply.py
Comment (trbromley) [reply tests/test_apply.py:1137]: :thinking: This one's still not clear to me - so before we had `analytic=True` and `shots=10000`. In the test, we call `dev.sample()` so it would have generated `10000` samples, yet these samples were not used in `dev.probability()` because we were in analytic mode?

So perhaps the test didn't need to set `shots` or call `dev.sample()`?

Would this just work if we had a parametrize over shots, setting `[None, 10000]`, so we test both analytic and non-analytic cases?

File: tests/test_expval.py
Comment (trbromley) [reply tests/test_expval.py:33]: Actually, do you agree that the `shots` fixture is not actively used in any of the tests below? :confused: Unless it's hidden somewhere in a fixture. So we could set to any value.

In which case, ideally we would remove the paramtrize and `shots` argument. But ok for this PR if we want to just change to `None` as you have done :+1:

**Review by trbromley — COMMENTED** [review]
Thanks @antalszava! Looks close to :heavy_check_mark:, I just had a few final questions.

File: pennylane_lightning/lightning_qubit.py
Comment (trbromley) [inline pennylane_lightning/lightning_qubit.py:43]: Also, we need to update `pennylane_requires = ">=0.12"` to the latest version

File: tests/test_expval.py
Comment (antalszava) [reply tests/test_expval.py]: Yep, removed it

File: tests/test_apply.py
Comment (antalszava) [reply tests/test_apply.py]: Sure, changed this. Also removed the calls to `dev.sample`.

**Review by trbromley — APPROVED** [review]
Thanks @antalszava for accommodating those suggestions! :100:

**Pull request merged**
